src/modules/authentification/components/AuthLoginForm.tsx

```diff
@@ -1,48 +1,138 @@
-import { useForm, FormProvider } from "react-hook-form";
-import InputField from "./inputs/InputField";
+import { auth, provider } from "../../firebase/firebase";
+import {
+  signInWithEmailAndPassword,
+  GoogleAuthProvider,
+  signInWithPopup,
+} from "firebase/auth";
 import {
   passwordMaxLength,
   passwordMinLength,
   passwordRegexp,
+  emailRegexp,
 } from "../const/userInputRequirements";
+import { userState, loggedState } from "../state/userState";
+import { useSetRecoilState } from "recoil";
+import { useNavigate, Link } from "react-router-dom";
+import googleLogoImg from "../assets/google-logo.png";
+import InputField from "./inputs/InputField";
+import { useForm, FormProvider } from "react-hook-form";
+import Button from "../../../shared/Button";
+
+interface LoginData {
+  email: string;
+  password: string;
+}
 
 export default function AuthLoginForm() {
-  const form = useForm();
+  const navigate = useNavigate();
+
+  const form = useForm<LoginData>();
+  const setUserData = useSetRecoilState(userState);
+  const setIsLoggedIn = useSetRecoilState(loggedState);
+
+  const onSubmit = (data: LoginData) => {
+    if (!data.email || !data.password) return;
```

Comment:
No need for this if you have input validation.

Reply:
Removed.